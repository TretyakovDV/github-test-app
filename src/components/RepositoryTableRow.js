import React from 'react';
import PropTypes from 'prop-types';
import { IconButton, TableCell, TableRow } from '@material-ui/core';
import DeleteIcon from '@material-ui/icons/Delete';
import LinkIcon from '@material-ui/icons/Link';
import SubjectIcon from '@material-ui/icons/Subject';

const RepositoryTableRow = ({
  id,
  title,
  stars,
  url,
  onDetails,
  onDelete,
}) => {
  const handleDetails = () => onDetails(id);
  const handleDelete = () => onDelete(id);

  return (
    <TableRow>
      <TableCell width={250}>{title}</TableCell>
      <TableCell width={100} align="right">{stars}</TableCell>
      <TableCell align="center">
        <IconButton onClick={handleDetails} data-testid="details">
          <SubjectIcon />
        </IconButton>
      </TableCell>
      <TableCell align="center">
        <IconButton component="a" href={url} target="_blank">
          <LinkIcon />
        </IconButton>
      </TableCell>
      <TableCell align="center">
        <IconButton onClick={handleDelete} data-testid="delete">
          <DeleteIcon />
        </IconButton>
      </TableCell>
    </TableRow>
  );
};

RepositoryTableRow.propTypes = {
  id: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  stars: PropTypes.number.isRequired,
  url: PropTypes.string.isRequired,
  onDetails: PropTypes.func.isRequired,
  onDelete: PropTypes.func.isRequired,
};

export default RepositoryTableRow;
