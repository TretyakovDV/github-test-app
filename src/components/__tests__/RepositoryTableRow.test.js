import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import RepositoryTableRow from '../RepositoryTableRow';

describe('RepositoryTableRow', () => {
  it('matches snapshot', () => {
    const id = 1;
    const title = 'title';
    const stars = 200;
    const url = 'url';

    const onDetails = jest.fn();
    const onDelete = jest.fn();

    const { asFragment } = render(
      <RepositoryTableRow
        id={id}
        title={title}
        stars={stars}
        url={url}
        onDetails={onDetails}
        onDelete={onDelete}
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('onDetails()', () => {
    const id = 1;
    const title = 'title';
    const stars = 200;
    const url = 'url';

    const onDetails = jest.fn();
    const onDelete = jest.fn();

    const { getByTestId } = render(
      <RepositoryTableRow
        id={id}
        title={title}
        stars={stars}
        url={url}
        onDetails={onDetails}
        onDelete={onDelete}
      />,
    );

    fireEvent.click(getByTestId('details'));

    expect(onDetails).toHaveBeenCalledTimes(1);
  });

  it('onDelete()', () => {
    const id = 1;
    const title = 'title';
    const stars = 200;
    const url = 'url';

    const onDetails = jest.fn();
    const onDelete = jest.fn();

    const { getByTestId } = render(
      <RepositoryTableRow
        id={id}
        title={title}
        stars={stars}
        url={url}
        onDetails={onDetails}
        onDelete={onDelete}
      />,
    );

    fireEvent.click(getByTestId('delete'));

    expect(onDelete).toHaveBeenCalledTimes(1);
  });
});
