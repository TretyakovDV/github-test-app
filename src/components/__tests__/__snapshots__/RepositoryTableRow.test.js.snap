// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoryTableRow matches snapshot 1`] = `
<DocumentFragment>
  <tr
    class="MuiTableRow-root"
  >
    <td
      class="MuiTableCell-root"
      width="250"
    >
      title
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-alignRight"
      width="100"
    >
      200
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-alignCenter"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M14 17H4v2h10v-2zm6-8H4v2h16V9zM4 15h16v-2H4v2zM4 5v2h16V5H4z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-alignCenter"
    >
      <a
        aria-disabled="false"
        class="MuiButtonBase-root MuiIconButton-root"
        href="url"
        tabindex="0"
        target="_blank"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1zM8 13h8v-2H8v2zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76 0 5-2.24 5-5s-2.24-5-5-5z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </a>
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-alignCenter"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </td>
  </tr>
</DocumentFragment>
`;
